cookbook/examples/workflows/procurement_analysis/workflow.py

```diff
@@ -0,0 +1,137 @@
+from agno.workflow.v2 import Condition, Parallel, Step, Workflow
+from agno.workflow.v2.types import StepInput
+
+from agents import (
+    alternative_suppliers_agent,
+    company_overview_agent,
+    cost_drivers_agent,
+    kraljic_agent,
+    pestle_agent,
+    porter_agent,
+    report_compiler_agent,
+    switching_barriers_agent,
+)
+from models import ProcurementAnalysisRequest
+
+
+def should_run_analysis(analysis_type: str) -> callable:
+    def evaluator(step_input: StepInput) -> bool:
+        request_data = step_input.message
+        if isinstance(request_data, ProcurementAnalysisRequest):
+            return analysis_type in request_data.analyses_requested
+        return False
+
+    return evaluator
+
+company_overview_step = Step(
+    name="Company Overview",
+    agent=company_overview_agent,
+    description="Research and analyze the target company",
+)
+
+switching_barriers_step = Step(
+    name="Switching Barriers Analysis",
+    agent=switching_barriers_agent,
+    description="Analyze supplier switching barriers and costs",
+)
+
+pestle_step = Step(
+    name="PESTLE Analysis",
+    agent=pestle_agent,
+    description="Conduct PESTLE analysis for procurement strategy",
+)
+
+porter_step = Step(
+    name="Porter's Five Forces Analysis",
+    agent=porter_agent,
+    description="Analyze competitive forces in the supply market",
+)
+
+kraljic_step = Step(
+    name="Kraljic Matrix Analysis",
+    agent=kraljic_agent,
+    description="Position category on Kraljic Matrix",
+)
+
+cost_drivers_step = Step(
+    name="Cost Drivers Analysis",
+    agent=cost_drivers_agent,
+    description="Analyze cost structure and volatility",
+)
+
+alternative_suppliers_step = Step(
+    name="Alternative Suppliers Research",
+    agent=alternative_suppliers_agent,
+    description="Identify and evaluate alternative suppliers",
+)
+
+report_compilation_step = Step(
+    name="Report Compilation",
+    agent=report_compiler_agent,
+    description="Compile comprehensive procurement analysis report",
+)
+
+procurement_workflow = Workflow(
+    name="Procurement Analysis Workflow",
+    description="Comprehensive procurement intelligence using multiple strategic frameworks",
+    steps=[
+        company_overview_step,
+        Parallel(
```

Comment:
Technically a parallel isn't needed here, because you only ever run one of the below right? In that case I would suggest Router instead of Parallel + Condition.